### ISSUE TITLE
gnome and xfce can't coexist due to conflicting desktopManagerHandlesLidAndPower

### ISSUE THREAD

**Issue by bdimcheff** [NixOS/nixpkgs#9641]
It seems that the recent changes (#9590) to desktopManagerHandlesLidAndPower may prevent gnome3 and xfce from being enabled simultaneously due to conflicting definitions of that variable.  I would assume that any time 2 desktop managers have different values here, they can't be enabled at the same time.

```
error: The option `services.xserver.displayManager.desktopManagerHandlesLidAndPower' has conflicting definitions, in `/path/to/nixpkgs/nixos/modules/services/x11/desktop-managers/gnome3.nix' and `/path/to/nixpkgs/nixos/modules/services/x11/desktop-managers/xfce.nix'.
(use ‘--show-trace’ to show detailed location information)
```

**Comment by vcunat** [comment]
See #9642 for discussion on solutions.

**Comment by bdimcheff** [comment]
I think this is fixed now, at least in unstable, no?

**Comment by vcunat** [comment]
Yes, it should be fixed on both unstable and 15.09 (to be announced soon).
